### PR TITLE
slims down kubermatic api docker image

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: slims down kubermatic api docker image

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,7 @@
...
+FROM scratch
 
 LABEL maintainer "sebastian@loodse.com"
 
-RUN echo https://mirror1.hs-esslingen.de/pub/Mirrors/alpine/v3.7/main > /etc/apk/repositories; \
-    echo https://mirror1.hs-esslingen.de/pub/Mirrors/alpine/v3.7/community >> /etc/apk/repositories
-
-RUN apk add --no-cache ca-certificates
-
-ADD https://storage.googleapis.com/kubernetes-release/release/v1.10.3/bin/linux/amd64/kubectl /usr/local/bin/kubectl
```
Comment (p0lyn0mial) [inline api/Dockerfile:10]: I think we don't need `kubectl` to be present in the image

File: api/Dockerfile [hunk 2]
```diff
@@ -1,15 +1,7 @@
-FROM alpine:3.7
+FROM scratch
```
Comment (p0lyn0mial) [inline api/Dockerfile:1]: because we can . . . :)

File: api/Dockerfile (hunk 1 above)
Comment (mrIncompetent) [reply api/Dockerfile:10]: We need kubectl, otherwise the addon-controller won't work anymore. It directly calls `kubectl`

Comment (alvaroaleman) [reply api/Dockerfile:10]: I think the addon-controller uses it

Comment (mrIncompetent) [reply api/Dockerfile:10]: 2 seconds :scream:

File: api/Dockerfile [hunk 3]
```diff
@@ -1,15 +1,7 @@
...
 
 LABEL maintainer "sebastian@loodse.com"
 
-RUN echo https://mirror1.hs-esslingen.de/pub/Mirrors/alpine/v3.7/main > /etc/apk/repositories; \
-    echo https://mirror1.hs-esslingen.de/pub/Mirrors/alpine/v3.7/community >> /etc/apk/repositories
-
-RUN apk add --no-cache ca-certificates
```
Comment (alvaroaleman) [inline api/Dockerfile:8]: I think the API needs the ca-certificates to validate the dex certificate

Comment (mrIncompetent) [reply api/Dockerfile:8]: Correct, when we pull certs via lets-encrypt we need the CA's to validate

**Comment by p0lyn0mial** [comment]
Thanks for your input :) I'm going to decline this PR.

**Pull request closed without merging**